backend/account/authentication_helper.py

```diff
@@ -43,6 +43,8 @@ def get_or_create_user_by_email(user_id: str, email: str) -> User:
         """
         user_service = UserService()
         user = user_service.get_user_by_email(email)
+        if user and not user.user_id:
+            user = user_service.update_user(user, user_id)
```

Comment:
Why do we do this update here in the `get_or_create()` method?

Reply:
This particular scenario is where we have user_id only when the user accepts the invite and logs in.
But at the same time we have to create user in a database for frictionless onboarded user.